lib/logstash/filters/translate.rb

```diff
@@ -128,23 +142,53 @@ def register
     end
   end # def register
 
-  public
-  def load_yaml(registering=false)
-    if !File.exists?(@dictionary_path)
-      @logger.warn("dictionary file read failure, continuing with old dictionary", :path => @dictionary_path)
+  private
+  def load_file(registering,fileName,yamlData=nil)
+    if !File.exists?(fileName)
+      @logger.warn("dictionary file read failure, continuing with old dictionary", :path => fileName)
       return
     end
 
     begin
-      @dictionary.merge!(YAML.load_file(@dictionary_path))
+      if yamlData==nil
+        yamlData = YAML.load_file(fileName)
+      end
+      @dictionary.merge!(yamlData)
     rescue Exception => e
       if registering
-        raise "#{self.class.name}: Bad Syntax in dictionary file #{@dictionary_path}"
+        raise "#{self.class.name}: Bad Syntax in dictionary file #{fileName}"
       else
-        @logger.warn("#{self.class.name}: Bad Syntax in dictionary file, continuing with old dictionary", :dictionary_path => @dictionary_path)
+        @logger.warn("#{self.class.name}: Bad Syntax in dictionary file, continuing with old dictionary", :dictionary_path => fileName)
+      end
+    end
+  end # def load_file
+
+  public
+  def load_yaml(registering=false)
+    load_file(registering,@dictionary_path)
+  end # def load_yaml
+
+  public
+  def download_yaml(path,registering=false)
+    filename = Digest::SHA1.hexdigest path;
+    File.open(filename+"_temp.yml", "wb") do |saved_file|
+      open(path, "rb") do |read_file|
+        saved_file.write(read_file.read)
       end
     end
-  end
+    rename = true
+    begin
+        yamlData = YAML.load_file(filename+"_temp.yml")
+    rescue Exception => e
+        rename = false
+    end
+    if rename
+        FileUtils.mv(filename+"_temp.yml", filename+".yml")
+    else
+        FileUtils.rm_rf(filename+"_temp.yml")
```

Comment:
I didn't read about the vulnerabilities and risks of rm_rf, sorry! now i'm changing to rm_f instead. 
Thanks!